Packages/com.unity.inputsystem/InputSystem/InputSystem.cs

```diff
@@ -3308,6 +3308,23 @@ public static int ListEnabledActions(List<InputAction> actions)
             return InputActionState.FindAllEnabledActions(actions);
         }
 
+        /// <summary>
+        /// Find an action by its ID (see <see cref="InputAction.id"/>).
+        /// </summary>
+        /// <param name="id">ID of the action to look for.</param>
+        /// <returns>Returns the action with the given ID or <see langword="null"/> if no action could be found with the given ID.</returns>
+        /// <remarks>
+        /// For the action to be found, the action must be part of an <see cref="InputActionAsset"/>
+        /// that has had any of its actions enabled at some point. For embedded actions that are not part of an asset, the
+        /// action must have been enabled at some point.
+        /// </remarks>
+        /// <seealso cref="InputActionAsset.FindAction(Guid)"/>
+        /// <seealso cref="InputActionMap.FindAction(Guid)"/>
+        public static InputAction FindAction(Guid id)
```

Comment:
I would prefer not to have this going looking into the runtime state.
As you put in the remarks there is a lot of caveats as to whether and when this function will work.
This exposes directly the innards of InputSystem which means the public API will now need to maintain this behaviour (e.g. Action must have been enabled etc). Changing ActionState might break that contract.

The other 'FindAction' methods are not using runtime state but rather the data registered with InputSystem (assets, action maps or whatever) that later feed into the runtime state. If this is really needed, then we would need to add some type of array `InputAction`s of registered single actions and aggregate that with action maps and actions, if we don't have it already.